File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int total = brown + yellow;
    
    for(int i = 3; i * i <= total; i++) {
        if(total % i != 0) continue;
        int j = total / i;
        if(yellow == (i - 2) * (j - 2)) {
            answer.push_back(i);
            answer.push_back(j);
            break;
        }
    }
    
    sort(answer.begin(), answer.end(), greater<>());
        
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
from math import sqrt

def solution(brown, yellow):
    answer = []
    
    total = brown + yellow
    for i in range(1, int(sqrt(total)) + 1):
        if total % i != 0: 
            continue
            
        j = int(total / i)
        if (i - 2) * (j - 2) == yellow:
            answer = [j, i]
            break
    
    return answer